Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -446,9 +446,9 @@ String HTMLInputElement::value_sanitization_algorithm(String value) const
         }
     } else if (type_state() == HTMLInputElement::TypeAttributeState::Number) {
         // If the value of the element is not a valid floating-point number, then set it to the empty string instead.
-        char* end_ptr;
-        auto val = strtod(value.characters(), &end_ptr);
-        if (!isfinite(val) || *end_ptr)
+        // FIXME: The spec does not specify if we need to trim whitespace.
```

Comment:
I think we can trim the whitespace here and in the HTMLProgressElement. I found this:

https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#rules-for-parsing-floating-point-number-values
Specifically step 6:
> [Skip ASCII whitespace](https://infra.spec.whatwg.org/#skip-ascii-whitespace) within input given position.

Also the **Example** box in https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-value mentions that a UA could trim whitespace on input elements in general if it wanted.

Reply:
Nice catch, found that spec but only looked for if the format was compatible, whitespace doesn't count in format right :facepalm:

I made them all trim the whitespace and added a link to the spec.